### PR TITLE
bug fixes for RNG AF quest line.

### PR DESCRIPTION
**Pull request by ProfessorCattington**
Small fixes to the RNG AF quest line;

Fixed cases where items weren't showing up in the chat log properly during events.
Fixed cases where events could fire or repeat before requirements were met.
Changed the 'Old Sabertooth' encounter so players on the correct quest step can only advance if they are in the cave with the mob when it dies.
Fixed the Scavenge script to report results properly to chat log. note: action:param() doesn't seem to behave the way the script originally assumed.

### CODE REVIEW

**Comment by teschnei** [comment]
Can you elaborate on your action:param note?

**Comment by ProfessorCattington** [comment]
Oh sorry,
In the original script for scavenge we call 6 setters on action. All of those calls had syntax errors so none of them worked. I updated the calls so the ability as a whole behaves as you would expect except for action:param(). In this case, I assume the function is for reporting what item the player found with the ability but it always returns a blank to the chat log. example;
scavengereport. Unless I'm misusing action:param I'm sure what the problem is and It doesn't look like we use this function for any other ability.

**Comment by teschnei** [comment]
oh yes, I rewrote the lua_action file and forgot to check if anything was using the old stuff.  Anyways, the action's param is probably overwritten by the return value.  So you can probably just remove the action:param and just return the arrow id, and I think that will work

**Comment by ProfessorCattington** [comment]
Looks like you were right. The ability no properly reports the item you found to the chat log.

File: scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua [hunk 1]
```diff
@@ -13,12 +13,35 @@ require("scripts/globals/status");
...
+
+        --check to see if the player's x,y,z coordinates are near enough to the cave where the RNG quest mob dies.
+        --point 1: 690, -15, -338
+        --point 2: 654, -9, -383
+        --these are outside the playable space but used to form a box around the cave where this quest takes place
+            local playerX = player:getXPos();
+            --local playerY = player:getYPos(); we shouldn't really care about the y axis for this
+            local playerZ = player:getZPos();
+
+            if (playerX < 690 and playerX > 654 and
```
Comment (teschnei) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: why not check the distance of the player instead? it's much more readable

Comment (ProfessorCattington) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: Do you mean distance from the a point within the cave? That seems reasonable assuming there's a vector magnitude function. If not I can probably just add it.

Comment (bendangelo) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: It should be `player:distance(x, y, z)`. Although I don't think it's implemented.

Comment (teschnei) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: you can probably just use the distance from the ??? (or even the tiger itself, maybe).

player:checkDistance(entity)

Comment (ProfessorCattington) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: I like these ideas. I will get at this when I have a chance.

**Comment by ProfessorCattington** [comment]
So I tried out and tested checkDistance. it seems to work fine but then I realized there's a small part of this quest missing. I believe on live this quest will not progress unless the mob dies without aggro. The whole theme of the quest is witnessing natural death rather than killing something. I'm not seeing any checkAggro or aggro status functions that will tell me whether or not a mob is currently in an aggro state. Does such a function exist already and if not would the next best thing be use something related like current TP?

**Comment by teschnei** [comment]
Don't do weird workarounds. If a function you need doesn't exist, ask for
it. There is an isengaged in the core, but I don't know if it's exposed and
whether or not it will work in onMobDeath. I'll try to remember to look
when I get home tonight

**Comment by ProfessorCattington** [comment]
Looked at this a moment ago. Not sure isEngaged will work since it's returning a animation state. It seems like putting an aggro flag on battleentity and turning it on or off in the OnEngage/OnDisengage then exposing that flag would be pretty harmless.

**Comment by teschnei** [comment]
No, we will not be adding another field to battleentity. I don't think a
byte for all ten thousand entities for one quest is gonna be that useful.
The most probable solution will be to check claim, I think. That's probably
what retail does.

**Comment by ProfessorCattington** [comment]
Do you mean this 'PMaster' field?

**Comment by teschnei** [comment]
just check if killer is nil or not - if it's not nil, then someone claimed it and killed it

**Comment by ProfessorCattington** [comment]
ah ok. thanks!

File: scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua [hunk 2]
```diff
@@ -13,12 +13,28 @@ require("scripts/globals/status");
...
 
 function onMobSpawn(mob)
+
 end;
 
 -----------------------------------
 -- onMobDeath
 -----------------------------------
 
-function onMobDeath(mob, killer, ally)
```
Comment (xdemolish) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: why did you remove ally

Comment (ProfessorCattington) [inline scripts/zones/Sauromugue_Champaign/mobs/Old_Sabertooth.lua]: That was probably just a mistake. I added back in.
